File: second/task.cpp
#include <iostream>
#include <vector>

int findWinner(int n, int k, int k2 = 0) {
    if (n == k2)
        return 1;
    k2 = k2 + 1;
    return (findWinner(n - 1, k), k2 + k - 1) % n + 1;
}

int main() {
    int n, k;
    std::cout << "Введите количество друзей (n): ";
    std::cin >> n;
    std::cout << "Введите число (k): ";
    std::cin >> k;

    int winner = findWinner(n, k);
    std::cout << "Победитель: друг под номером " << winner << " (рекурсивный метод)" << std::endl;

    return 0;
}

File: second/task.go
package main

import "fmt"

func findWinner(n, k int) int {
	if n == 1 {
		return 1
	}
	return (findWinner(n-1, k)+k-1)%n + 1
}

func main() {
	var n, k int
	fmt.Print("Введите количество друзей (n): ")
	fmt.Scan(&n)
	fmt.Print("Введите число (k): ")
	fmt.Scan(&k)

	winner := findWinner(n, k)
	fmt.Printf("Победитель: друг под номером %d\n", winner)
}
